pkg/stanza/fileconsumer/internal/archive/archive_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package archive // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/archive"
```

Comment:
One of the main benefits of having a separate package is that we can have more focused tests. The two tests below seem to evaluate a couple particular nuances of how the archive works but it's not clear to me that we are covering the basics. Test the constructor, including edge cases. Test CRUD operations...

I suggest changing the package name in this file to `archive_test` to ensure the tests are focused on the exported behaviors of `archive`.

Reply:
> One of the main benefits of having a separate package is that we can have more focused tests. The two tests below seem to evaluate a couple particular nuances of how the archive works but it's not clear to me that we are covering the basics. Test the constructor, including edge cases. Test CRUD operations...

I've added tests to cover these scenarios. PTAL when you can. Thanks!

Reply:
> I suggest changing the package name in this file to archive_test to ensure the tests are focused on the exported behaviors of archive.

If we do this, we will no longer be able to test the type assertions such as https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/44245a3461dac49c5a2547669f044585aec17c44/pkg/stanza/fileconsumer/internal/archive/archive_test.go#L37-L41

The [`archive`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/44245a3461dac49c5a2547669f044585aec17c44/pkg/stanza/fileconsumer/internal/archive/archive.go#L46-L54) struct is unexported and is a part of `archive` package. https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/44245a3461dac49c5a2547669f044585aec17c44/pkg/stanza/fileconsumer/internal/archive/archive.go#L46-L54

Reply:
> If we do this, we will no longer be able to test the type assertions

That's ok though. We should validate based on the expected behavior of the exported elements only. e.g. Validate by calling `WriteFiles` and then `FindFiles` to see if it is actually doing the work or just a nop implementation.

Reply:
Got it. I have made necessary adjustments.

Reply:
I will continue to reiterate this because I see only incremental progress in this direction.

We should change the test package name. This will force us to test only the externally observable behaviors.

For example, I still see access of internal elements, such as validating the expected position of the archive, or loading archives directly with checkpoint. The purpose of the package is to abstract a problem away from the users of it. Tests should simulate usage, not validate implementation details.

```suggestion
package archive_test
```

Reply:
> I will continue to reiterate this because I see only incremental progress in this direction.

@djaglowski  Crap, I forgot to push this change. Sorry, my bad.

> For example, I still see access of internal elements, such as validating the expected position of the archive, or loading archives directly with checkpoint. The purpose of the package is to abstract a problem away from the users of it. Tests should simulate usage, not validate implementation details.

I agree. I'll push my changes with changes in package name.

Reply:
@djaglowski It should be good now. I've removed all references to internal details.